src/Installers/Windows/AspNetCoreModule-Setup/ANCMIISExpressV2/AncmIISExpressV2.wixproj

```diff
@@ -70,5 +71,27 @@
 
   <PropertyGroup>
     <PackageFileName>ancm_iis_express_$(Platform)_en_v2_$(PackageVersion)$(TargetExt)</PackageFileName>
+    <ProductNameShort>ASP.NET Core Module IIS Express V2</ProductNameShort>
   </PropertyGroup>
+
+  <Target Name="CreateANCMIISExpressNugetPackage" AfterTargets="CopyToArtifactsDirectory;Build">
```

Comment:
This is just copied from existing code: https://github.com/dotnet/aspnetcore/blob/fb537bb78353b74df428b9ec3974c496ee964828/src/Installers/Windows/SharedFramework/SharedFramework.wixproj#L89. But like you said, `CopyToArtifactsDirectory` runs _before_ `Build`. We want this to run after both - if I remove `Build` it could run between `CopyToArtifactsDirectory` and `Build`.

Reply:
To run "after" both, use `BeforeTargets="Build"` because `Build` _should_ be an empty target that just marks where building ends and because `AfterTargets` targets sometimes don't get run
```suggestion
  <Target Name="CreateANCMIISExpressNugetPackage" AfterTargets="CopyToArtifactsDirectory" BeforeTargets="Build">
```

Reply:
So to run after `Build`, I have to set `Build` as a `BeforeTarget`? That doesn't make sense - is that an msbuild bug?

Reply:
And could you elaborate on "`AfterTargets` targets sometimes don't get run"?

Reply:
Sorry, took a bit to respond because I wanted to confirm `Build` was empty for WiX projects. Preprocessed output doesn't show the target at all and my local build from root wasn't working. Fortunately, I got what I need from a binary log: The microsoft.signed.wix package defines `Build` and it is indeed empty.

> That doesn't make sense - is that an msbuild bug?

Not a bug because `Build` doesn't do anything; it's just used as a place to hang the real "build" work and as a marker.

> And could you elaborate on "AfterTargets targets sometimes don't get run"?

There are sometimes cleanup issues that mean processing ends before `AfterTargets` get run. But more importantly, `msbuild` doesn't guarantee anything about _when_ after the named target(s) what you're defining runs. That can cause unforeseen issues e.g. if pack or test operations run before your `AfterTargets="Build"` target. At least that's how I remember it.

The generally recommended pattern is to have an empty "marker" target like `Build` that clearly delineates where `Build` work ends and use `BeforeTargets="Build"` (or whatever target marks the ending event you want). The only real "gotcha" is making sure your target depends on the other targets it needs, especially if their ordering is slightly ambiguous.

Note I should have used `DependsOnTargets="CopyToArtifactsDirectory"` in my final suggestion because `AfterTargets` doesn't guarantee the named target(s) will ever run (`AfterTargets` just says "run me if that other gal or gals run"); `DependsOnTargets` explicitly says "if you run me, make sure this other gal or gals has run first".

@rainersigwald anything to add❔

Reply:
Separately, the current setup could mean the `CreateANCMIISExpressNugetPackage` runs just once after `CopyToArtifactsDirectory` because `msbuild` will detect it already ran when `Build` is reached.

Reply:
Updated: https://github.com/dotnet/aspnetcore/pull/39140/commits/3c2a22761fac12af193d4655c3ca4e46aa256889